packages/contracts-bedrock/test/Preinstalls.t.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { Preinstalls } from "src/libraries/Preinstalls.sol";
+
+interface IEIP712 {
+    function DOMAIN_SEPARATOR() external view returns (bytes32);
+}
+
+/// @title PredeploysTest
+contract PredeploysTest is CommonTest {
+    /// @dev The domain separator commits to the chainid of the chain
+    function test_preinstall_permit2_succeeds() external {
+        bytes32 domainSeparator = IEIP712(Preinstalls.PERMIT2).DOMAIN_SEPARATOR();
+        bytes32 typeHash =
+            keccak256(abi.encodePacked("EIP712Domain(string name,uint256 chainId,address verifyingContract)"));
+        bytes32 nameHash = keccak256(abi.encodePacked("Permit2"));
+        uint256 chainId = block.chainid;
```

Comment:
Can we make this a fuzz test over chain ID? We'd have to re-run the genesis script as part of the test but that should be ok

Reply:
[Updated](https://github.com/ethereum-optimism/optimism/pull/9317/files#diff-23eeac34b4924ed4ff89fab11a236d05c0f12eb6f7b62f21c242a19ab79dec6cR14-R24)!

Reply:
This is just testing the value returned by `DOMAIN_SEPARATOR()`, which [is recomputed](https://github.com/Uniswap/permit2/blob/cc56ad0f3439c502c246fc5cfcc3db92bb8b7219/src/EIP712.sol#L26-L30) (i.e. does not use the immutable) if the chain ID changed. So this test confirms that `DOMAIN_SEPARATOR()` works as intended, but does not test that we've set the correct immutables based on chain ID